### PR TITLE
[tests] create a test for public inputs

### PR DESCRIPTION
**Pull request by mimoo**
will write a PR description in 3, 2, 1, ...

closes https://github.com/o1-labs/proof-systems/issues/356

### CODE REVIEW

File: kimchi/src/circuits/polynomials/generic.rs [hunk 1]
```diff
@@ -330,9 +331,14 @@ pub mod testing {
...
                     * this[register_offset + 1];
+                let public = if coeffs_offset == 0 {
+                    public.get(row).cloned().unwrap_or(F::zero())
+                } else {
+                    F::zero()
+                };
                 ensure_eq!(
                     zero,
-                    sum + mul + self.coeffs[coeffs_offset + 4],
+                    sum + mul + self.coeffs[coeffs_offset + 4] - public,
```
Comment (mrmr1993) [inline kimchi/src/circuits/polynomials/generic.rs:341]: I don't think this is right: the public input is already accounted for as part of the witness, and the gates for the public input should already be at the beginning of the witness. The old version was already correct.

**Review by mrmr1993 — CHANGES_REQUESTED** [review]
Only the updates to the test framework are right here; the rest (changes to `verify` functions) should be reverted

**Comment by mimoo** [comment]
sorry didn't see you were reviewing, the tests still don't pass so I'm looking for what's wrong

File: kimchi/src/circuits/polynomials/generic.rs (hunk 1 above)
Comment (mimoo) [reply kimchi/src/circuits/polynomials/generic.rs:341]: I'm not sure I understand. If you don't remove the public element, which is in the witness, then the sum is not going to be equal to zero.

For the first `public` rows the equation is going to be `generic_selector( alpha^gen1 * (1 * w[0])) = w[0]`

Comment (mimoo) [reply kimchi/src/circuits/polynomials/generic.rs:341]: indeed the tests pass now, so that must be that I'm right :P

Comment (mrmr1993) [reply kimchi/src/circuits/polynomials/generic.rs:341]: Ah, indeed. My mistake!

**Review by mrmr1993 — APPROVED** [review]

**Pull request merged**